<!--REQUEST_ID=123b1cc0-6eb9-409a-aad2-a999274e94bd-->
Failure on aarch64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.50873&attempt_id=24b41ed3-acf7-4ddf-916b-082c38cbb15b)

Attempted: openiscsi

<details><summary>Partial log (click to expand)</summary><p>

```
/build/source/usr/iscsid.c:355: undefined reference to `sd_notifyf'
iscsid.o: In function `set_state_to_ready':
/build/source/usr/iscsid.c:347: undefined reference to `sd_notify'
/build/source/usr/iscsid.c:343: undefined reference to `sd_notify'
collect2: error: ld returned 1 exit status
make[1]: *** [Makefile:68: iscsid] Error 1
make[1]: Leaving directory '/build/source/usr'
make: *** [Makefile:53: user] Error 2
builder for '/nix/store/fhj9cy6a2g63l6qs1z3ra0il4g1in9br-open-iscsi-2.0.877.drv' failed with exit code 2
error: build of '/nix/store/fhj9cy6a2g63l6qs1z3ra0il4g1in9br-open-iscsi-2.0.877.drv' failed
```
</p></details>

--- a/pkgs/os-specific/linux/open-iscsi/default.nix
+++ b/pkgs/os-specific/linux/open-iscsi/default.nix
@@ -1,26 +1,30 @@
-{ stdenv, fetchFromGitHub, automake, autoconf, libtool, gettext, utillinux, openisns, openssl, kmod }:
+{ stdenv, fetchFromGitHub, automake, autoconf, libtool, gettext
+, utillinux, openisns, openssl, kmod, perl, systemd, pkgconf
+}:
+
 stdenv.mkDerivation rec {
   name = "open-iscsi-${version}";
-  version = "2.0-873-${stdenv.lib.substring 0 7 src.rev}";
+  version = "2.0.877";
 
-  buildInputs = [ automake autoconf libtool gettext utillinux openisns.lib openssl kmod ];
-  
+  nativeBuildInputs = [ autoconf automake gettext libtool perl pkgconf ];
+  buildInputs = [ kmod openisns.lib openssl systemd utillinux ];
+
   src = fetchFromGitHub {
     owner = "open-iscsi";
     repo = "open-iscsi";
-    rev = "4c1f2d90ef1c73e33d9f1e4ae9c206ffe015a8f9";
-    sha256 = "0h030zk4zih3l8z5662b3kcifdxlakbwwkz1afb7yf0cicds7va8";
+    rev = version;
+    sha256 = "0v3dsrl34pdx0yl5jsanrpgg3vw466rl8k81hkshgq3a5mq5qhf6";
   };
-  
+
   DESTDIR = "$(out)";
-  
-  NIX_LDFLAGS = "-lkmod";
+
+  NIX_LDFLAGS = "-lkmod -lsystemd";
   NIX_CFLAGS_COMPILE = "-DUSE_KMOD";
 
   preConfigure = ''
     sed -i 's|/usr|/|' Makefile
   '';
-  
+
   postInstall = ''
     cp usr/iscsistart $out/sbin/
     $out/sbin/iscsistart -v
@@ -28,9 +32,9 @@ stdenv.mkDerivation rec {
 
   meta = with stdenv.lib; {
     description = "A high performance, transport independent, multi-platform implementation of RFC3720";
-    license = licenses.gpl2Plus;
-    homepage = http://www.open-iscsi.com;
+    license = licenses.gpl2;
+    homepage = https://www.open-iscsi.com;
     platforms = platforms.linux;
-    maintainers = with maintainers; [ cleverca22 ];
+    maintainers = with maintainers; [ cleverca22 zaninime ];
   };
 }

